In [1]:
import seaborn as sns
titanic = sns.load_dataset("titanic")
titanic.isna().sum()

survived         0
pclass           0
sex              0
age            177
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

In [5]:
titanic

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,embark_town,alive,alone
0,0,3,male,22.0,1,0,7.2500,S,Third,man,True,Southampton,no,False
1,1,1,female,38.0,1,0,71.2833,C,First,woman,False,Cherbourg,yes,False
2,1,3,female,26.0,0,0,7.9250,S,Third,woman,False,Southampton,yes,True
3,1,1,female,35.0,1,0,53.1000,S,First,woman,False,Southampton,yes,False
4,0,3,male,35.0,0,0,8.0500,S,Third,man,True,Southampton,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,S,Second,man,True,Southampton,no,True
887,1,1,female,19.0,0,0,30.0000,S,First,woman,False,Southampton,yes,True
888,0,3,female,21.5,1,2,23.4500,S,Third,woman,False,Southampton,no,False
889,1,1,male,26.0,0,0,30.0000,C,First,man,True,Cherbourg,yes,True


In [2]:
nb_age_manquants = titanic["age"].isna().sum()

mediane_age = titanic.groupby(["class", "sex"], observed=True)["age"].transform("median")
titanic["age"] = titanic["age"].fillna(mediane_age)

titanic.isna().sum()

survived         0
pclass           0
sex              0
age              0
sibsp            0
parch            0
fare             0
embarked         2
class            0
who              0
adult_male       0
deck           688
embark_town      2
alive            0
alone            0
dtype: int64

In [3]:
nb_port_manquants = titanic["embark_town"].isna().sum()
port_le_plus_frequent = titanic["embark_town"].mode()[0]
titanic["embark_town"] = titanic["embark_town"].fillna(port_le_plus_frequent)

titanic = titanic.drop(columns="deck")

titanic.isna().sum()

survived       0
pclass         0
sex            0
age            0
sibsp          0
parch          0
fare           0
embarked       2
class          0
who            0
adult_male     0
embark_town    0
alive          0
alone          0
dtype: int64

In [4]:
nb_doublons = titanic.duplicated().sum()
nb_doublons

np.int64(118)

In [6]:
titanic[titanic.duplicated(keep=False)].sort_values(list(titanic.columns))

,survived,pclass,sex,age,sibsp,parch,fare,embarked,class,who,adult_male,embark_town,alive,alone
263,0,1,male,40.0,0,0,0.0000,S,First,man,True,Southampton,no,True
633,0,1,male,40.0,0,0,0.0000,S,First,man,True,Southampton,no,True
815,0,1,male,40.0,0,0,0.0000,S,First,man,True,Southampton,no,True
30,0,1,male,40.0,0,0,27.7208,C,First,man,True,Cherbourg,no,True
64,0,1,male,40.0,0,0,27.7208,C,First,man,True,Cherbourg,no,True
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
692,1,3,male,25.0,0,0,56.4958,S,Third,man,True,Southampton,yes,True
65,1,3,male,25.0,1,1,15.2458,C,Third,man,True,Cherbourg,yes,False
709,1,3,male,25.0,1,1,15.2458,C,Third,man,True,Cherbourg,yes,False
74,1,3,male,32.0,0,0,56.4958,S,Third,man,True,Southampton,yes,True


In [7]:
titanic = titanic.drop(columns=["embarked", "class", "alive", "who", "adult_male"])
titanic["sex"] = titanic["sex"].astype("category")
titanic["embark_town"] = titanic["embark_town"].astype("category")

titanic.dtypes

survived          int64
pclass            int64
sex            category
age             float64
sibsp             int64
parch             int64
fare            float64
embark_town    category
alone              bool
dtype: object

In [8]:
titanic

,survived,pclass,sex,age,sibsp,parch,fare,embark_town,alone
0,0,3,male,22.0,1,0,7.2500,Southampton,False
1,1,1,female,38.0,1,0,71.2833,Cherbourg,False
2,1,3,female,26.0,0,0,7.9250,Southampton,True
3,1,1,female,35.0,1,0,53.1000,Southampton,False
4,0,3,male,35.0,0,0,8.0500,Southampton,True
...,...,...,...,...,...,...,...,...,...
886,0,2,male,27.0,0,0,13.0000,Southampton,True
887,1,1,female,19.0,0,0,30.0000,Southampton,True
888,0,3,female,21.5,1,2,23.4500,Southampton,False
889,1,1,male,26.0,0,0,30.0000,Cherbourg,True


In [9]:
titanic["fare"].describe()

count    891.000000
mean      32.204208
std       49.693429
min        0.000000
25%        7.910400
50%       14.454200
75%       31.000000
max      512.329200
Name: fare, dtype: float64

In [10]:
q1 = titanic["fare"].quantile(0.25)
q3 = titanic["fare"].quantile(0.75)
borne_haute = q3 + 1.5 * (q3 - q1)

billets_hors_bornes = titanic[titanic["fare"] > borne_haute].sort_values("fare", ascending=False)
nb_hors_bornes = len(billets_hors_bornes)
print(f"Borne haute : {borne_haute:.2f} / Billets hors bornes : {nb_hors_bornes}")
billets_hors_bornes

Borne haute : 65.63 / Billets hors bornes : 116


,survived,pclass,sex,age,sibsp,parch,fare,embark_town,alone
258,1,1,female,35.0,0,0,512.3292,Cherbourg,True
737,1,1,male,35.0,0,0,512.3292,Cherbourg,True
679,1,1,male,36.0,0,1,512.3292,Cherbourg,False
27,0,1,male,19.0,3,2,263.0000,Southampton,False
341,1,1,female,24.0,3,2,263.0000,Southampton,False
...,...,...,...,...,...,...,...,...,...
324,0,3,male,25.0,8,2,69.5500,Southampton,False
369,1,1,female,24.0,0,0,69.3000,Cherbourg,True
641,1,1,female,24.0,0,0,69.3000,Cherbourg,True
336,0,1,male,29.0,1,0,66.6000,Southampton,False


## Journal de nettoyage

| Quoi | Pourquoi | Lignes |
|---|---|---|
| deck supprimée | 77 % de valeurs manquantes | 891 |
| age imputé | médiane de la classe et du sexe | 177 |
| embark_town imputé | valeur la plus fréquente, seulement 2 cas | 2 |
| doublons conservés | pas d'identifiant passager, deux lignes identiques peuvent être deux personnes différentes | 118 |
| colonnes redondantes supprimées | répètent d'autres colonnes (embarked, class, alive, who, adult_male) | 891 |
| types corrigés | sex et embark_town passés en category | 891 |
| billets hors bornes conservés | vrais billets de première classe, pas des erreurs de saisie | 116 |